### PR TITLE
[TIMOB-20337] Parity: enabling run-on-main-thread

### PR DESCRIPTION
**Pull request by cheekiatng**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-20337

### CODE REVIEW

**Comment by FokkeZB** [comment]
Shouldn't we still support and deprecate the old way first?

**Comment by cheekiatng** [comment]
It's ok. When it was introduced earlier, it was already mentioned that it's experimental so there should not be any project dependencies on this.

**Comment by hansemannn** [comment]
Would be cool to at least throw a log warning once and remove that for 6.0.0 so everyone know's what's going on.

**Comment by cheekiatng** [comment]
Sounds reasonable. Will update pr

File: node_modules/titanium-sdk/tests/resources/tiapp2.xml [hunk 1]
```diff
@@ -35,6 +34,7 @@
 	<property name="ti.bb.invoke.target.key.push" type="string">ti.testapp.invoke.push</property>
 	<property name="ti.bb.invoke.target.key.open" type="string">ti.testapp.invoke.open</property>
 	<property name="push_title" type="string">Some Title for BB Push, typically the app name"</property>
+	<property name="run-on-main-thread" type="bool">true</property>
 	<code-processor>
```
Comment (jhaynie) [inline node_modules/titanium-sdk/tests/resources/tiapp2.xml]: yeah :+1:

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -2076,7 +2076,14 @@ iOSBuilder.prototype.initialize = function initialize() {
 	// Remove the debugHost/profilerHost check when we have debugging/profiling support with JSCore framework
 	// TIMOB-17892
 	this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && (this.tiapp.ios['use-jscore-framework'] || false);
-	this.currentBuildManifest.runOnMainThread = this.runOnMainThread = (this.tiapp['run-on-main-thread'] === true);
+	// Remove this check on 6.0.0
+	if(this.tiapp.ios && (this.tiapp.ios.hasOwnProperty('run-on-main-thread'))) {
```
Comment (feons) [inline iphone/cli/commands/_build.js]: there should a space between `if` and `(`

File: iphone/cli/commands/_build.js [hunk 3]
```diff
@@ -2076,7 +2076,14 @@ iOSBuilder.prototype.initialize = function initialize() {
...
+		this.logger.info(__('run-on-main-thread no longer set in the <ios> section of the tiapp.xml. Use <property name="run-on-main-thread" type="bool">true</property> instead'));
+		this.currentBuildManifest.runOnMainThread = this.runOnMainThread = (this.tiapp.ios['run-on-main-thread'] === true);
+	}
+	else {
```
Comment (feons) [inline iphone/cli/commands/_build.js]: `else` should not go on the next line.

File: node_modules/titanium-sdk/lib/tiappxml.js [hunk 4]
```diff
@@ -99,7 +95,10 @@ function toXml(dom, parent, name, value) {
 			if (value.hasOwnProperty('use-jscore-framework')) {
 				dom.create('use-jscore-framework', { nodeValue: !!value['use-jscore-framework'] }, node);
 			}
-
+ 
+ 			if (value.hasOwnProperty('run-on-main-thread')) {
+ 				dom.create('run-on-main-thread', { nodeValue: !!value['run-on-main-thread'] }, node);
+ 			}
```
Comment (feons) [inline node_modules/titanium-sdk/lib/tiappxml.js]: space here just to be consistent.

**Comment by feons** [comment]
Looks good. **Approved.**

**Pull request merged**